.github/workflows/release.yml

```diff
@@ -0,0 +1,120 @@
+        uses: actions/checkout@v4
+
+      # Validate Wrapper before running build
+      - name: validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@56b90f209b02bf6d1deae490e9ef18b21a389cd4 #v1.1.0
+
+      # Set up Java environment for the next steps
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+
+      # Setup Gradle
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-home-cache-cleanup: true
+
+      - name: Set Release Version
+        id: release_version
+        shell: bash
+        run: |
+          CURRENT_VERSION=$(grep "projectVersion=" gradle.properties | cut -d'=' -f2)
+          NEW_VERSION=${CURRENT_VERSION%-SNAPSHOT}
+          awk -v current="$CURRENT_VERSION" -v new="$NEW_VERSION" 'BEGIN {FS=OFS="="} $1 == "projectVersion" { $2 = new }1' gradle.properties > tmpfile && mv tmpfile gradle.properties
+          echo "Release version: $NEW_VERSION"
+          echo "PLUGIN_VERSION=${NEW_VERSION}" >> $GITHUB_ENV
+
+      # Publish the plugin to JetBrains Marketplace
+      - name: Publish Plugin to JetBrains Marketplace
+        env:
+          PUBLISH_TOKEN: ${{ secrets.JETBRAINS_MARKETPLACE_TOKEN }}
+        run: |
+          ./gradlew publishPlugin -PjetBrainsToken=$PUBLISH_TOKEN -PprojectVersion=$PLUGIN_VERSION -PjetBrainsChannel=stable
+          echo "Published $PLUGIN_VERSION to the Jetbrains Marketplace"
+
+      - name: Tag Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config user.email "action@github.com"
+          git config user.name "GitHub Action"
+          if git diff --quiet; then
+            echo "No changes to commit."
+          else
+            git commit -sam "chore(skip-release): set version to $PLUGIN_VERSION"
+          fi
+          git tag $PLUGIN_VERSION
+          git push origin $PLUGIN_VERSION
+
+      # Set next SNAPSHOT version
+      # TODO uncomment this when CI ( prow/integration tests ) is stable enough, or bypass github settings
+      #- name: Increment Plugin Version
+      #  env:
+      #    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      #  run: |
+      #    CURRENT_VERSION=$(grep "projectVersion=" gradle.properties | cut -d'=' -f2)
+      #    IFS="-" read -ra VERSION_PARTS <<< "$CURRENT_VERSION"
+      #    IFS="." read -ra VERSION_NUM <<< "${VERSION_PARTS[0]}"
+      #    ((VERSION_NUM[2]++))
+      #    NEW_VERSION="${VERSION_NUM[0]}.${VERSION_NUM[1]}.${VERSION_NUM[2]}-SNAPSHOT"
+      #    awk -v new_version="$NEW_VERSION" '/projectVersion=/{sub(/=.*/, "=" new_version)}1' gradle.properties > tmpfile && mv tmpfile gradle.properties
+      #    echo "Set $NEW_VERSION in gradle.properties"
+      #    git commit -sam "chore(skip-release): set version to $NEW_VERSION"
+      #    git push origin main
+
+      - name: Simple conventional changelog
```

Comment:
here  we're using conventional changelog that you're not using to generate changelog. this must be changed to what you're currenlty doing, OR must use a check to be sure every commit as a correct changelog titile

Reply:
@sbouchet: I added the check: `conventionalCheck.yml`. Maybe I should file this in a separate PR that we merge first?

Reply:
yes, that makes sense to do this way.

Reply:
@sbouchet: added the semantic commit check in a separate commit for this PR